### DISCUSSION TITLE
nfsmount -flat.vmdk image howto?

### DISCUSSION THREAD

**Discussion by jonsito** [ltsp/ltsp#849] [Q&A]
Hello all,

I'm using Virtualbox to generate client images. It works fine either via NBD or NFS with squashfs generated images from flat.vmdk virtualbox file

Aside note: NFS performance is by far large better than NBD when working with multiple simultaneous clients and large squashfs image files. 

Surfing in code I've found ( file 55-images.sh in function mount_image_src() ) some coments saying that it is possible to export and use directly vmdk image without conversion to squashfs

My vbox image is properly flat.vmdk formatted with /  (ext4) as partition #1 and /boot/efi (vfat) as #2. 

My grub file says as attached image
Client starts, but fails trying to mount partition #1 from vmdk image (as shown in attached image). It seems as ltsp code does not know how to handle partitions on vmdk image and makes 'losetup' fail... perhaps a missing "-P" or "--partscan" in losetup command?

What am I doing wrong? Any ideas?
Thanks in advance
![Captura de pantalla de 2023-08-02 11-41-16](https://github.com/ltsp/ltsp/assets/828257/12e2633f-7c7a-48d3-83e5-66a464f02f7c)

![Captura de pantalla de 2023-08-02 11-48-47](https://github.com/ltsp/ltsp/assets/828257/56dc45da-bb84-4122-a26f-7041278d7ae0)

**Comment by alkisg** [comment]
What's the output of this command?

    sudo fdisk -l /path/to/your/vm-flat.vmdk

**Comment by jonsito** [comment]
Here comes. 
Anyway, adding "loop.part_max=9 partition=1" to grub.cfg entry as suggested did de trick

root@maestro:/data/maestro3-nfs-imagenes/clientes# LANG=C fdisk -l LABDit-2022-flat.vmdk
Disk LABDit-2022-flat.vmdk: 32 GiB, 34359738368 bytes, 67108864 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x7015e6fa

Device                 Boot   Start      End  Sectors  Size Id Type
LABDit-2022-flat.vmdk1      1052672 67106815 66054144 31,5G 83 Linux
LABDit-2022-flat.vmdk2 *       2048  1050623  1048576  512M  b W95 FAT32

Partition table entries are not in disk order.
